File: python/718. 最长重复子数组.py
class Solution_718:
    def findLength(self,nums1,nums2):
        m,n=len(nums1),len(nums2)
        if(m==0 or n==0):
            return 0
        dp=[[0]*(n+1) for _ in range(m+1)]
        ret=0
        for i in range(1,m+1):
            for j in range(1,n+1):
                if(nums1[i-1]==nums2[j-1]):
                    dp[i][j]=dp[i-1][j-1]+1
                    ret=max(dp[i][j],ret)
        return  ret
if __name__ == '__main__':
    nums1=eval(input("数组nums1"))
    nums2=eval(input("数组nums2"))
    s_718=Solution_718()
    print(s_718.findLength(nums1,nums2))

File: c++/718. 最长重复子数组.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;
class Solution_718 {
	public:
		int findLength(vector<int>& nums1, vector<int>& nums2) {
			int m = nums1.size(), n = nums2.size();
			if (m == 0 || n == 0) return 0;
			vector<vector<int>> dp(m + 1, vector<int>(n + 1));
			int ret = 0;
			for (int i = 1; i <= m; ++i) {
				for (int j = 1; j <= n; ++j) {
					if (nums1[i - 1] == nums2[j - 1]) {
						dp[i][j] = dp[i - 1][j - 1] + 1;
						ret = max(dp[i][j], ret);
					}
				}
			}
			return ret;
		}
};
void main_718() {
	vector<int> nums1;
	vector<int> nums2;
	int num;
	while (cin >> num) {
		nums1.push_back(num);
		if (cin.get() == '\n') break;
	}
	while (cin>>num){
		nums2.push_back(num);
		if (cin.get() == '\n') break;
	}
	Solution_718 s_718;
	cout<<s_718.findLength(nums1, nums2)<<endl;
}